gnovm/pkg/gnolang/values.go

```diff
@@ -724,12 +725,19 @@ func (mv *MapValue) GetLength() int {
 // is not returned as part of the pointer, and TV is not filled.
 func (mv *MapValue) GetPointerForKey(alloc *Allocator, store Store, key *TypedValue) PointerValue {
 	// If NaN, instead of computing map key, just append to List.
-	kmk := key.ComputeMapKey(store, false)
-	if mli, ok := mv.vmap[kmk]; ok {
-		return PointerValue{
-			TV:    fillValueTV(store, &mli.Value),
-			Base:  mv,
-			Index: PointerIndexMap,
+	kmk, isNaN := key.ComputeMapKey(store, false)
+	if !isNaN {
+		if mli, ok := mv.vmap[kmk]; ok {
+			key2 := key.Copy(alloc)
```

Comment:
can you please check this PR: https://github.com/gnolang/gno/pull/4327. 
it looks `key` should always be copied?

Reply:
Okay, sounds good. I'll apply that change here with you as co-author.